packages/flutter/lib/src/material/drop_down.dart

```diff
@@ -533,10 +538,15 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> {
           mainAxisAlignment: MainAxisAlignment.spaceBetween,
           mainAxisSize: MainAxisSize.min,
           children: <Widget>[
-            new IndexedStack(
-              index: _selectedIndex,
-              alignment: FractionalOffset.centerLeft,
-              children: config.items
+            // The button's size is defined by its largest menu item. If value is
+            // null then an item does not appear.
+            new Opacity(
+              opacity: _selectedIndex == null ? 0.0 : 1.0,
+              child: new IndexedStack(
+                index: _selectedIndex ?? 0,
```

Comment:
Maybe you should teach indexedstack to take a null index, which means "show none of the children"